File: backend/internal/domain/grade.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Grade struct {
	ID        uuid.UUID `db:"id"`
	GroupID   uuid.UUID `db:"group_id"`
	StudentID uuid.UUID `db:"student_id"`
	TeacherID uuid.UUID `db:"teacher_id"`
	Subject   string    `db:"subject"`
	Value     float64   `db:"value"`
	MaxValue  float64   `db:"max_value"`
	Comment   *string   `db:"comment"`
	GradedAt  time.Time `db:"graded_at"`
	CreatedAt time.Time `db:"created_at"`
}

File: flutter_app/lib/features/grades/domain/entities/grade.dart
import 'package:flutter/foundation.dart';

@immutable
class Grade {
  const Grade({
    required this.id,
    required this.groupId,
    required this.studentId,
    required this.teacherId,
    required this.subject,
    required this.value,
    required this.maxValue,
    required this.gradedAt,
    required this.createdAt,
    this.comment,
  });

  final String id;
  final String groupId;
  final String studentId;
  final String teacherId;
  final String subject;
  final double value;
  final double maxValue;
  final String? comment;
  final DateTime gradedAt;
  final DateTime createdAt;

  double get percentage => maxValue > 0 ? value / maxValue * 100 : 0;
}
